src/command_modules/azure-cli-batchai/azure/cli/command_modules/batchai/custom.py

```diff
@@ -31,31 +40,88 @@
                                 'provide storage_account value in batchai section of your az configuration file.'
 MSG_CONFIGURE_STORAGE_KEY = 'Please configure Azure Storage account key via AZURE_BATCHAI_STORAGE_KEY or ' \
                             'provide storage_key value in batchai section of your az configuration file.'
+STANDARD_OUTPUT_DIRECTORY_ID = 'stdouterr'
 
 # Placeholders which customer may use in his config file for cluster creation.
 AZURE_BATCHAI_STORAGE_KEY_PLACEHOLDER = '<{0}>'.format(AZURE_BATCHAI_STORAGE_KEY)
 AZURE_BATCHAI_STORAGE_ACCOUNT_PLACEHOLDER = '<{0}>'.format(AZURE_BATCHAI_STORAGE_ACCOUNT)
 
+# Default expiration time for file download URLs.
+DEFAULT_URL_EXPIRY_MIN = 60
+
 # Supported images.
-SUPPORTED_IMAGES = {
-    "ubuntults": models.ImageReference(
+SUPPORTED_IMAGE_ALIASES = {
+    "UbuntuLTS": models.ImageReference(
         publisher='Canonical',
         offer='UbuntuServer',
         sku='16.04-LTS'
     ),
-    "ubuntudsvm": models.ImageReference(
+    "UbuntuDSVM": models.ImageReference(
         publisher='microsoft-ads',
         offer='linux-data-science-vm-ubuntu',
         sku='linuxdsvmubuntu'
     )
 }
 
+# Type of entries reported by list startup files.
+LogFile = collections.namedtuple('LogFile', 'name download_url is_directory size')
+
+logger = get_logger(__name__)
+
+
+def _get_resource_group_location(cli_ctx, resource_group):
+    client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES)
+    return client.resource_groups.get(resource_group).location
+
+
+def _get_default_ssh_public_key_location():
+    path = os.path.join(os.path.expanduser('~'), '.ssh', 'id_rsa.pub')
+    if os.path.exists(path):
+        return path
+    return None
+
 
 def _get_deserializer():
     client_models = {k: v for k, v in models.__dict__.items() if isinstance(v, type)}
     return Deserializer(client_models)
 
 
+def _ensure_resource_not_exist(client, resource_group, name):
```

Comment:
Why is this here? Create commands are expected to be idempotent. This would prevent that.

Reply:
Because we had a lot's of complains from the customers like "i submitting new job but it's not executing because i have already executed a job with such name o month ago" and asks to fix it

Reply:
Let's let this in if it's helpful to customers. Part of my action items will be to have a discussion on our side to see if we can recommend a more helpful and consistent solution for these scenarios.